src/main/java/kg/jarkyn/server/RequestingSocket.java

```diff
@@ -0,0 +1,33 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class RequestingSocket implements Requester {
```

Comment:
I think that name is misleading. That's your actual (client) socket. But it's not only requesting. Technically it is receiving data, too.
Or will for that matter, because `getOutputStream()` currently returns `null`. ;)